string_permutation_problem/permute.py

```diff
@@ -0,0 +1,23 @@
+def permute_my_string(str):
+    output_string = []
+    """
+        This is a recursive solution to the string permutation problem.
+        The user is supposed to return all the permutations for a given string.
+    """
+    #Base Case
+    if len(str) == 1:
+        output_string = [str]       
+    else:
+        #For every letter in String
+        for i, letter in enumerate(str):
+            #for every permutation resulting from Step 2 and 3 described above
+            for perm in permute(str[:i] + str[i+1:]):
+                #Add it to output
```

Comment:
Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `python spacing`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp16gmzyf8/string_permutation_problem/permute.py
+++ b/tmp/tmp16gmzyf8/string_permutation_problem/permute.py
@@ -6,7 +6,7 @@
     """
     #Base Case
     if len(str) == 1:
-        output_string = [str]       
+        output_string = [str]
     else:
         #For every letter in String
         for i, letter in enumerate(str):
```